### ISSUE TITLE
Invalid method setForceIndexLabel for ml_* algorithms called with formula

### ISSUE THREAD

**Issue by javierluraschi** [rstudio/sparklyr#1113]
This one is specific to Spark 1.6.X and might be a regression from `CRAN` release...

```r

sc <- spark_connect(master = "local", version = "1.6.1")

sex_df <- data.frame(
    sex = c("male", "male", "male", "male", "female", "female", "female", "female"),
    height = c(6, 5.92, 5.58, 5.92, 5, 5.5, 5.42, 5.75),
    weight = c(180, 190, 170, 165, 100, 150, 130, 150),
    foot = c(12, 11, 12, 10, 6, 8, 7, 9)
)
sex_tbl <- copy_to(sc, sex_df)

ml_naive_bayes(sex_tbl, formula = sex ~ height + weight + foot)
```

```java
Error: java.lang.IllegalArgumentException: invalid method setForceIndexLabel for object 60
	at sparklyr.Invoke$.invoke(invoke.scala:122)
	at sparklyr.StreamHandler$.handleMethodCall(stream.scala:97)
	at sparklyr.StreamHandler$.read(stream.scala:62)
	at sparklyr.BackendHandler.channelRead0(handler.scala:52)
	at sparklyr.BackendHandler.channelRead0(handler.scala:14)
	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:105)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:308)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:294)
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:308)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:294)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:244)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:308)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:294)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:846)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:131)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:511)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:468)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:382)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:354)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:111)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:137)
	at java.lang.Thread.run(Thread.java:748) 
```

**Comment by kevinykuo** [comment]
Related: https://stackoverflow.com/questions/47189528/sparklyr-ml-linear-regression-error-invalid-method-setforceindexlabel.

We're calling a method of `RFormula` that was implemented in 2.1.0. I'll work on a fix. Current workaround is to use Spark 2.1.0+

**Comment by kevinykuo** [comment]
Fixed in https://github.com/rstudio/sparklyr/pull/1126
